lib/takeaway.rb

```diff
@@ -0,0 +1,68 @@
+
+require_relative 'text.rb'
+
+class Takeaway
+
+  attr_reader :menu, :order_as_hash, :customer_supplied_total, :actual_total,
+  :text
+
+  def initialize(text_klass = Text.new)
+    @menu = {"starter" => 5, "mains" => 10, "desert" => 3}
+    @text = text_klass
+  end
+
+  def show_dishes
+    menu.each do |key, value|
+    puts "#{key} : #{value}"
+    end
+  end
+
+  def order(*dishes)
+    get_customer_total(dishes)
+    check_order(dishes)
+    text_confirmation(text)
+  end
+
+
+
+  private
+
+  def get_customer_total(dishes)
+    @customer_supplied_total = dishes.pop
+  end
+
+  def check_order (dishes)
```

Comment:
Style/SpaceAfterMethodName: Do not put a space between a method name and the opening parenthesis. (https://github.com/bbatsov/ruby-style-guide#parens-no-spaces)